pkgs/development/libraries/openssl/default.nix

```
@@ -152,13 +152,8 @@ let
       rmdir $out/etc/ssl/{certs,private}
     '';
 
-    postFixup = lib.optionalString (!stdenv.hostPlatform.isWindows) ''
-      # Check to make sure the main output doesn't depend on perl
-      if grep -r '${buildPackages.perl}' $out; then
-        echo "Found an erroneous dependency on perl ^^^" >&2
-        exit 1
-      fi
-    '';
+    disallowedReferences = [ buildPackages.perl ]
+      ++ lib.optional (!withPerl) perl;
```

Would this make sense?
```suggestion
    disallowedReferences = lib.optional ((buildPackages.perl != perl) || !withPerl) [ buildPackages.perl ];
```

---

Not exactly; you always don't want build perl (assuming it's different from runtime perl), but you also don't want runtime perl if !withPerl